colour/algebra/interpolation.py

```diff
@@ -31,23 +37,598 @@
+
+        Parameters
+        ----------
+        value : int
+            Value to set the half window with.
+
+        Returns
+        -------
+        int
+            Half window.
+        """
+
+        return self._window
+
+    @window.setter
+    def window(self, value):
+        """
+        Setter for the **self.window** property.
+        """
+
+        if value is not None:
+            assert is_integer(value), '"window" must be an integer!'
+
+            assert value >= 1, '"window" must be equal or superior to 1!'
+
+            self._window = value
+
+            if self._x is not None:
+                self.x = self._x
+
+            if self._y is not None:
+                self.y = self.y
+
+    @property
+    def kernel(self):
+        """
+        Getter and setter property for the kernel callable.
+
+        Parameters
+        ----------
+        value : callable
+            Value to set the kernel callable.
+
+        Returns
+        -------
+        callable
+            Kernel callable.
+        """
+
+        return self._kernel
+
+    @kernel.setter
+    def kernel(self, value):
+        """
+        Setter for the **self.kernel** property.
+        """
+
+        if value is not None:
+            assert hasattr(
+                value,
+                '__call__'), ('"{0}" attribute: "{1}" is not callable!'.format(
+                    'kernel', value))
+
+            self._kernel = value
+
+    @property
+    def kernel_args(self):
+        """
+        Getter and setter property for the kernel call time arguments.
+
+        Parameters
+        ----------
+        value : dict
+            Value to call the interpolation kernel with.
+
+        Returns
+        -------
+        dict
+            Kernel call time arguments.
+        """
+
+        return self._kernel_args
+
+    @kernel_args.setter
+    def kernel_args(self, value):
+        """
+        Setter for the **self.kernel_args** property.
+        """
+
+        if value is not None:
+            assert isinstance(value, (dict, OrderedDict)), (
+                '"{0}" attribute: "{1}" type is not "dict" or "OrderedDict"!'
+            ).format('kernel_args', value)
+
+            self._kernel_args = value
+
+    @property
+    def padding_args(self):
+        """
+        Getter and setter property for the kernel call time arguments.
+
+        Parameters
+        ----------
+        value : dict
+            Value to call the interpolation kernel with.
+
+        Returns
+        -------
+        dict
+            Kernel call time arguments.
+        """
+
+        return self._padding_args
+
+    @padding_args.setter
+    def padding_args(self, value):
+        """
+        Setter for the **self.padding_args** property.
+        """
+
+        if value is not None:
+            assert isinstance(value, (dict, OrderedDict)), (
+                '"{0}" attribute: "{1}" type is not "dict" or "OrderedDict"!'
+            ).format('padding_args', value)
```

Comment:
Does it have to be a `dict` or `OrderedDict` wouldn't any mapping type do?

Reply:
Good point, I think checking for `collections.Mapping` should be enough, I'll update the check.